### PR TITLE
Add forge template to `scarb new` and `scarb init` under separate flag

### LINKED ISSUE #1208: Add forge template to `scarb new` under separate flag.
### Problem

As in title

### Proposed Solution

_No response_

### Notes

_No response_

### PR DESCRIPTION
**Pull request by Draggu**
Closes  #1208

- Added `--snforge` flag that enables forge template for
   - `new` command
   - `init` command
   
  On forge side https://github.com/foundry-rs/starknet-foundry/pull/2113

### CODE REVIEW

File: scarb/src/bin/scarb/args.rs [hunk 1]
```diff
@@ -241,6 +241,10 @@ pub struct InitArgs {
     /// Do not initialize a new Git repository.
     #[arg(long)]
     pub no_vcs: bool,
+
+    /// Initialize snforge setup.
```
Comment (mkaput) [inline scarb/src/bin/scarb/args.rs:245]:
```suggestion
    /// Use a Starknet Foundry Forge template.
```

image

File: scarb/src/bin/scarb/args.rs [hunk 2]
```diff
@@ -241,6 +241,10 @@ pub struct InitArgs {
     /// Do not initialize a new Git repository.
     #[arg(long)]
     pub no_vcs: bool,
```
Comment (maciektr) [inline scarb/src/bin/scarb/args.rs:243]: This args would be ignored, right?

File: scarb/src/ops/new.rs [hunk 3]
```diff
@@ -20,6 +21,7 @@ pub struct InitOptions {
     pub path: Utf8PathBuf,
     pub name: Option<PackageName>,
     pub vcs: VersionControl,
```
Comment (maciektr) [inline scarb/src/ops/new.rs:23]: `name` and `vcs` args would be ignored?

File: scarb/src/ops/new.rs [hunk 4]
```diff
@@ -193,6 +199,19 @@ fn mk(
         "#})
     }
 
+    if snforge {
+        init_snforge(name)?;
```
Comment (maciektr) [inline scarb/src/ops/new.rs:203]: I am not sure if we want to both create the package here with Scarb and then with Snforge. Will Snforge overwrite what has been created by Scarb? It does call Scarb `new` once again under the hood, is it ok?

File: scarb/src/ops/new.rs [hunk 5]
```diff
@@ -193,6 +199,19 @@ fn mk(
...
+    }
+
+    Ok(())
+}
+
+fn init_snforge(name: PackageName) -> Result<()> {
+    Command::new("snforge")
```
Comment (maciektr) [inline scarb/src/ops/new.rs:210]: Shouldn't this inherit stdout/stderr? 
Shouldn't this get all env vars usually set by Scarb for subcommands (like `SCARB` so snforge does not call any other Scarb binary) - see https://github.com/software-mansion/scarb/blob/61dd36af1d28d60499832f06b3a5c9e83ba7647a/scarb/src/subcommands.rs#L11

File: scarb/src/ops/new.rs
[the remark this answers is not in the document]
Comment (Draggu) [reply scarb/src/ops/new.rs:1]: https://github.com/foundry-rs/starknet-foundry/pull/2113

File: scarb/src/ops/new.rs (hunk 4 above)
Comment (Draggu) [reply scarb/src/ops/new.rs]: Changed forge behaviour in https://github.com/foundry-rs/starknet-foundry/pull/2113

File: scarb/src/ops/new.rs (hunk 3 above)
Comment (Draggu) [reply scarb/src/ops/new.rs:24]: `name` is used and `vcs` is ignored by forge

**Review by mkaput — APPROVED** [review]

File: scarb/src/ops/new.rs (hunk 3 above)
Comment (maciektr) [reply scarb/src/ops/new.rs:24]: If we allow to use both `--no-vcs` and `--snforge` simultaneously - we should support it.

Comment (mkaput) [reply scarb/src/ops/new.rs:24]: I believe snforge is NOT touching vcs stuff; and this code is handling VCS by itself

Comment (maciektr) [reply scarb/src/ops/new.rs:24]: Ok, cool, makes sense!

**Review by maciektr — APPROVED** [review]
I am not sure about the logic on snforge side (expressed in comment in the forge repo). 

This side look good to go!

**Pull request merged**